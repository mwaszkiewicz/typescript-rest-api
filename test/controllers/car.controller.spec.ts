/* eslint-disable no-undef */
import express, { Application } from 'express';
import request from 'supertest';
import { pino } from 'pino';
import { CarService } from '../../src/services/car.service';
import CarController from '../../src/controllers/car.controller';

describe('car.controller', () => {
  let service: CarService;
  let app: Application;

  beforeAll(async () => {
    const logger = pino();
    service = new CarService(logger);

    const controller = new CarController(service, logger);
    app = express();
    app.use(express.json());
    app.use('/', controller.router);
  });

  it('create car endpoint', async () => {
    const result = await request(app).post('/api/v1/cars').send({ vin: '12345', brand: 'Audi', model: 'A4' });

    expect(result.body).toMatchSnapshot();
    expect(result.status).toEqual(201);
  });

  it('update car endpoint', async () => {
    const result = await request(app).patch('/api/v1/cars').send({ vin: '12345', brand: 'Audi', model: 'A4' });

    expect(result.body).toMatchSnapshot();
    expect(result.status).toEqual(200);
  });

  it('get all cars endpoint', async () => {
    const result = await request(app).get('/api/v1/cars');

    expect(result.body).toMatchSnapshot();
    expect(result.status).toEqual(200);
  });

  it('delete car endpoint', async () => {
    const result = await request(app).delete('/api/v1/cars/12345');

    expect(result.body).toMatchSnapshot();
    expect(result.status).toEqual(200);
  });

});
