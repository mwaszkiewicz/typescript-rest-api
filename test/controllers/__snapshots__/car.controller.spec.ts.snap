// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`car.controller create car endpoint 1`] = `
Object {
  "car": Object {
    "brand": "Audi",
    "model": "A4",
    "vin": "12345",
  },
  "message": "Car created",
}
`;

exports[`car.controller delete car endpoint 1`] = `
Object {
  "message": "Removed car: 1",
}
`;

exports[`car.controller get all cars endpoint 1`] = `
Object {
  "result": Array [
    Object {
      "brand": "Audi",
      "model": "A4",
      "vin": "12345",
    },
    Object {
      "brand": "BMW",
      "model": "X3",
      "vin": "65432",
    },
  ],
}
`;

exports[`car.controller update car endpoint 1`] = `
Object {
  "message": "Car updated: true",
}
`;
